### ISSUE TITLE
MPI build broken with Python 3.13

### ISSUE THREAD

**Issue by paugier** [h5py/h5py#2523]
In https://github.com/h5py/h5py/blob/5c4871e37a08d7b030dc6c18df21f45cd714e6c6/setup.py#L51C1-L51C68, there is

```
    SETUP_REQUIRES.append("mpi4py ==3.1.6; python_version>='3.12'")
```

However, I think mpi4py 3.1.6 is not compatible with Python 3.13 (compile error).

I was able to compile mpi4py 4.0.1 on Python 3.13.

It would be interesting to know if h5py is compatible with mpi4py 4...

**Comment by tacaswell** [comment]
Those are only used when `pip` / setuptools makes an isolated environment, you can use what ever version of `mpi4py` you have installed via `pip install h5py --no-build-isolation` (but you do have to get all of the build requirements installed).

**Comment by paugier** [comment]
Yes, there is a workaround and it's useful to check if h5py is compatible with mpi4py 4.

However, it's still necessary to fix the isolated build for Python 3.13.

Good news, this works:

```
pip install pkgconfig cython setuptools numpy
export CPATH=/bxfs/softwares/hdf5/1.14.2.mpi-gcc.13.2.0-openmpi.4.1.5/include
pip cache remove h5py; HDF5_MPI="ON" CC=mpicc pip install --no-binary=h5py h5py --no-build-isolation
pip install pytest pytest-mpi
mpirun -np 2 python -c 'import h5py; h5py.run_tests()'
```
